src/polymer/content-to-slot-usages.ts

```diff
@@ -172,27 +121,219 @@ class ContentToSlot extends HtmlRule {
         warnings.push(warning);
       }
     }
+    return warnings;
   }
 }
 
-// NOTE(rictic): This only works for very, very simple selectors. Do something
-//   smarter here.
-function simpleSelectorToPredicate(simpleSelector: string): dom5.Predicate {
-  simpleSelector = simpleSelector.trim();
-  const pieces = simpleSelector.split(',');
-  if (pieces.length > 1) {
-    return p.OR(...pieces.map(simpleSelectorToPredicate));
+/**
+ * Describes how what elements to add a certain slot attribute to, when
+ * migrating usages of a certain element from content to slot.
+ */
+interface SlotMigrationDescriptor {
+  /** If an element matches this predicate... */
+  predicate: dom5.Predicate;
+  /** ... it should be given a slot attribute with this as its value. */
+  slot: string;
+}
+
+/**
+ * Returns a description of how to migrate the children of a given element to
+ * distribute using slots rather than the shadow dom v0 content system. This
+ * assumes that the given element has already been migrated to slots, and is
+ * either statically known to the linter (see configuration below), or uses the
+ * `old-content-selector` attribute to explain how it used to do distribution.
+ */
+function getMigrationDescriptors(tagName: string, document: Document):
+    ReadonlyArray<SlotMigrationDescriptor>|undefined {
+  const [element, ] = document.getFeatures(
+      {kind: 'element', id: tagName, imported: true, externalPackages: true});
+  if (element) {
+    const result = determineMigrationDescriptors(element);
+    // If we can determine descriptors dynamically, return those.
+    if (result.success && result.value.length > 0) {
+      return result.value;
+    }
   }
-  if (simpleSelector[0] === '.') {
-    return p.hasClass(simpleSelector.slice(1));
+
+  // Otherwise, try to get the descriptors from our hardcoded knowledge of
+  // elements.
+  return staticConfig.get(tagName);
+}
+
+class DescriptorError {
+  constructor(public readonly message: string, public readonly slot: Slot) {
   }
-  if (simpleSelector.startsWith('[') && simpleSelector.endsWith(']')) {
-    return p.hasAttr(simpleSelector.slice(1, -1));
+}
+
+type Result<Good, Bad> = {
```

Comment:
..and the Ugly? 😁

Reply:
:P

actually… when you've got a good result, a recoverable error, and an unrecoverable error… not a bad idea